pkg/cmd/infra/deployer/deployer_test.go

```diff
@@ -197,7 +198,7 @@ func TestDeployer_deployScenarios(t *testing.T) {
 }
 
 func mkdeployment(version int64, status deployapi.DeploymentStatus) *kapi.ReplicationController {
-	deployment, _ := deployutil.MakeDeployment(deploytest.OkDeploymentConfig(version), kapi.Codecs.LegacyCodec(deployapi.SchemeGroupVersion))
+	deployment, _ := deployutil.MakeDeployment(deploytest.OkDeploymentConfig(version), kapi.Codecs.LegacyCodec(deployv1.SchemeGroupVersion))
```

Comment:
@kargakis I noticed we were using the wrong codec everywhere (and with recent changes in Kube encoding to internal is not allowed).  The end result of these changes is that `TestHandle_changeWithTemplateDiff` is failing (it actually triggers a deployment now), which may mean the test was always broken.  Can your PTAL?